search/src/search.rs

```diff
@@ -181,37 +190,63 @@ impl Search {
         Ok(self.highlight(snippet))
     }
 
-    fn get_fuzzy_preview(&self, term: &str, distance: &u8, body: String) -> Result<String> {
-        let pindex: HashMap<&str, Vec<(u32, u32)>> = HashMap::from([
-            ("flooding", vec![(2, 10)])
-        ]);
-        let (matched_term, start, end) = self.get_fuzzy_match(term, distance, pindex);
+    fn get_fuzzy_preview(
+        &self,
+        path: &str,
+        term: &str,
+        distance: &u8,
+        body: &str,
+    ) -> Result<String> {
+        let pindex: PageIndex = self
+            .index
+            .page_index(path)
+            .map_err(|_| SearchError("".to_string()))?;
+        let (matched_term, start, end) = self
+            .get_fuzzy_match(term, distance, pindex)
+            .map_err(|_| SearchError("".to_string()))?;
         // Another safe way to get substrings using char_indices
-        let start = body.char_indices().nth(start as usize).unwrap_or((0, ' ')).0;
-        let end = body.char_indices().nth(end as usize).unwrap_or((body.len()-1, ' ')).0;
-        let substring = &body[start..end];
-        Ok(substring.to_string().replace(&matched_term, &format!("**{}**", term)))
+        let start = body
+            .char_indices()
+            .nth(start.saturating_sub(20) as usize)
+            .unwrap_or((0, ' '))
+            .0;
+        let end = body
+            .char_indices()
+            .nth((end + 20) as usize)
+            .unwrap_or((body.len() - 1, ' '))
+            .0;
+        let substring = &format!("...{}...", &body[start..end]);
+        let substring = substring
+            .to_string()
+            .replace(&matched_term, &format!("**{}**", matched_term));
+        Ok(substring.replace('\n', " "))
     }
 
-    fn get_fuzzy_match(&self, term: &str, distance: &u8, pindex: HashMap<&str, Vec<(u32, u32)>>) -> (String, u32, u32) {
+    fn get_fuzzy_match(
```

Comment:
What we should handle in this function or downstream is the case where the "match" has a distance equal to the maximum of the lengths of the term and the "match". In this case they really don't have to do anything with one another, and there shouldn't be a match.

Reply:
Sorry, I don't understand what you mean

Reply:
The strings in the vectors indicated in the test-cases, are strings found *near* the searched term, that are expected to be contained in the preview (not the expected matches themselves).

Reply:
Cleaning up the testcases was moved to a new issue: https://github.com/georgbuechner/litt/issues/76